src/writer.rs

```diff
@@ -340,6 +341,18 @@ impl<W: Write> Writer<W> {
 pub struct ElementWriter<'a, W> {
     writer: &'a mut Writer<W>,
     start_tag: BytesStart<'a>,
+    state: AttributeIndent,
+}
+
+/// Track indent inside elements state
+#[derive(Clone, Copy, Eq, PartialEq)]
+pub enum AttributeIndent {
+    /// Initial state
+    NoneAttributesWritten(usize),
+    /// Keep indent that should be used if `new_line()` would be called
+    SomeAttributesWritten(usize),
+    /// Write specified indent before writing attribute in `with_attribute()`
+    Indent(usize),
```

Comment:
Please put it before `ElementWriter` (so that `struct` and `impl` blocks was not breaked by other type) and no need to make it public, it is used only internally

Reply:
Done!